Sources/Widgets/Widgets3D/ResliceCursorWidget/cprBehavior.js

```diff
@@ -0,0 +1,116 @@
+  const superClass = { ...publicAPI };
+
+  model.stretchPlaneName = 'Y';
+  model.stretchPlaneViewType = planeNameToViewType[model.stretchPlaneName];
+  model.crossPlaneName = 'Z';
+  model.crossPlaneViewType = planeNameToViewType[model.crossPlaneName];
+  model.previousData = null;
+
+  publicAPI.getActiveInteraction = () => {
+    if (
+      model.widgetState
+        .getStatesWithLabel(`lineIn${model.stretchPlaneName}`)
+        .includes(model.activeState)
+    ) {
+      return InteractionMethodsName.TranslateCenterAndUpdatePlanes;
+    }
+    if (
+      model.widgetState
+        .getStatesWithLabel(`lineIn${model.crossPlaneName}`)
+        .includes(model.activeState) ||
+      model.widgetState
+        .getStatesWithLabel(`rotationIn${model.crossPlaneName}`)
+        .includes(model.activeState)
+    ) {
+      return InteractionMethodsName.RotateLine;
+    }
+    return null;
+  };
+
+  publicAPI.handleLeftButtonPress = (callData) => {
+    model.previousData = callData;
+    return superClass.handleLeftButtonPress(callData);
+  };
+
+  publicAPI[InteractionMethodsName.TranslateCenterAndUpdatePlanes] = (
+    calldata
+  ) => {
+    const manipulator =
+      model.activeState?.getManipulator?.() ?? model.manipulator;
+
+    const { worldPosition, worldAxes } = manipulator.handleEvent(
+      model.previousData,
+      calldata,
+      model._apiSpecificRenderWindow
+    );
+    model.previousData = calldata;
+    publicAPI.updateCenterAndPlanes(worldPosition, worldAxes);
+  };
+
+  publicAPI.updateCenterAndPlanes = (worldPosition, worldAxes) => {
+    // Update center
+    const newBoundedCenter = publicAPI.getBoundedCenter(worldPosition);
+    model.widgetState.setCenter(newBoundedCenter);
+
+    // Update planes if axes are given
+    if (worldAxes) {
+      const getAxis = (idx) =>
+        vec3.normalize([], worldAxes.slice(3 * idx, 3 * idx + 3));
+      const planes = model.widgetState.getPlanes();
+      Object.keys(planes).forEach((viewType) => {
+        switch (Number.parseInt(viewType, 10)) {
+          case ViewTypes.YZ_PLANE:
+            planes[viewType] = {
+              normal: getAxis(0),
+              viewUp: getAxis(2),
+            };
+            break;
+          case ViewTypes.XZ_PLANE:
+            planes[viewType] = {
+              normal: getAxis(1),
+              viewUp: getAxis(2),
+            };
+            break;
+          case ViewTypes.XY_PLANE:
+            planes[viewType] = {
+              normal: getAxis(2),
+              viewUp: getAxis(1),
+            };
+            break;
+          default:
+            break;
+        }
+      });
+    }
+
+    updateState(
+      model.widgetState,
+      model._factory.getScaleInPixels(),
+      model._factory.getRotationHandlePosition()
+    );
+  };
+
+  publicAPI.translateCenterOnPlaneDirection = (nbSteps) => {
+    const stretchState = model.widgetState.getStatesWithLabel(
+      `lineIn${model.stretchPlaneName}`
+    )?.[0];
```

Comment:
I guess, scrolling should be performed on the manipulator of the widget and not on the manipulator of a "line" handle.
It should then be handled in the superclass behavior.

Reply:
I need to handle the scroll in the cprManipulator because it holds the `currentDistance`

Reply:
The manipulator of the widget can be a CPRManipulator. (and the manipulator of the line handle is also a CPRManipulator... and maybe they are the same instances...)